Extension/src/main.ts

```diff
@@ -58,27 +65,58 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
 
     // Notify users if debugging may not be supported on their OS.
     util.checkDistro(info);
+    await checkVsixCompatibility();
+    UpdateInsidersAccess();
 
     const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
     if (settings.intelliSenseEngine === "Disabled") {
         languageServiceDisabled = true;
-        disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
-            if (!reloadMessageShown && settings.intelliSenseEngine !== "Disabled") {
+    }
+    let currentIntelliSenseEngineValue: string | undefined = settings.intelliSenseEngine;
+    disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
+        const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
+        if (!reloadMessageShown && settings.intelliSenseEngine !== currentIntelliSenseEngineValue) {
+            if (currentIntelliSenseEngineValue === "Disabled") {
+                // If switching from disabled to enabled, we can continue activation.
+                currentIntelliSenseEngineValue = settings.intelliSenseEngine;
+                LanguageServer.activate();
+            } else {
+                // We can't deactivate or change engines on the fly, so prompt for window reload.
                 reloadMessageShown = true;
                 util.promptForReloadWindowDueToSettingsChange();
             }
-        }));
-    } else {
-        disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
-            if (!reloadMessageShown && settings.intelliSenseEngine === "Disabled") {
-                reloadMessageShown = true;
-                util.promptForReloadWindowDueToSettingsChange();
+        }
+    }));
+
+    if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) {
+        for (let i: number = 0; i < vscode.workspace.workspaceFolders.length; ++i) {
+            const config: string = path.join(vscode.workspace.workspaceFolders[i].uri.fsPath, ".vscode/c_cpp_properties.json");
+            if (await util.checkFileExists(config)) {
+                const doc: vscode.TextDocument = await vscode.workspace.openTextDocument(config);
+                vscode.languages.setTextDocumentLanguage(doc, "jsonc");
             }
-        }));
+        }
     }
-    await LanguageServer.activate();
 
-    UpdateInsidersAccess();
+    taskProvider = vscode.tasks.registerTaskProvider(CppBuildTaskProvider.CppBuildScriptType, cppBuildTaskProvider);
```

Comment:
cleanup: we have access to `disposables` here, so instead of tracking this in its own variable, we can add it to the `disposables` array.